mllib/src/main/scala/org/apache/spark/ml/param/params.scala

```diff
@@ -44,8 +45,14 @@ import org.apache.spark.ml.util.Identifiable
  *                See [[ParamValidators]] for factory methods for common validation functions.
  * @tparam T param value type
  */
-class Param[T](val parent: String, val name: String, val doc: String, val isValid: T => Boolean)
-  extends Serializable {
+class Param[T: ClassTag](
```

Comment:
maybe a litter simpler

```suggestion
class Param[T](
    val parent: String, val name: String, val doc: String, val isValid: T => Boolean
)(implicit paramValueClassTag: ClassTag[T]) extends Serializable
```

Reply:
Sounds good.

Reply:
@zhengruifeng 

I recall I tried this approach `(implicit paramValueClassTag: ClassTag[T])` before , but it make us hard to get the classTag object. So I prefer current approach.

Reply:
then that is fine